src/main/connect/exchange_msg.rs

```diff
@@ -46,6 +48,7 @@ impl<UID: Uid> ExchangeMsg<UID> {
         cm: ConnectionMap<UID>,
         our_pk: PublicEncryptKey,
         shared_key: SharedSecretKey,
+        our_addrs: HashSet<SocketAddr>,
```

Comment:
There is no use sending local IPs for ext-reachability. So prune the list before hand.

Also call these `our_global_direct_listeners` as noted elsewhere too.